### ISSUE TITLE
Manage Soft Credits: Contact field should be required

### ISSUE THREAD

**Issue by chrismcmahon** [SalesforceFoundation/Cumulus#2408]
Off the Opportunity record, on the Manage Soft Credits page click "Add another soft credit". 

Add a valid Contact with other fields empty and attempt to save the record. Note that attempting to save a record with no values for ROLE NAME/TYPE/AMOUNT yields error messages.

Attempt to save a soft credit record with no value for Contact. No error message appears regardless of whether ROLE NAME/TYPE/AMOUNT exist. 

Save a soft credit with no Contact results in simply a no-op: no record is generated on the Opportunity, no error messages are presented to the user. Suggest Contact should be required along with every other field on the new soft credit record.

(I found this by accidentally erasing my Contact just before Save. I was surprised at the result.)

**Comment by judisohn** [comment]
@chrismcmahon Normally I'd be 1000% with you on this, but interestingly enough this mirrors the behavior when you add a standard Contact Role record. If you select a role value but don't lookup to a Contact and save, you end up back on the Opportunity having done nothing. Since I presume there's absolutely nothing we can do there, should we handle the Manage Soft Credits interface differently? I'm not sure.

**Comment by davidhabib** [comment]
because the page contains multiple rows in which to enter data, the presence or absence of a contact is what it currently uses to dictate whether that row should be committed.

if we feel this isn't good enough, then the change would be to validate/commit the row if ANY field in the row has data.

**Comment by judisohn** [comment]
@davidhabib Where did we end up on this? It's not Lurched.

**Comment by davidhabib** [comment]
I don't believe we discussed or resolved so I'll lurch it to make sure we have the discussion.

**Comment by davidhabib** [comment]
**lurch: add

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x80000005kji5AAA" target="_blank">W-018265</a>

**Comment by mbeaty-sf** [comment]
W-7781129

**Comment by jesslopez-sf** [comment]
Please follow this [Known Issue](https://trailblazer.salesforce.com/issues_view?id=a1p4V000000phvEQAQ&title=manage-soft-credits-contact-field-should-be-required) in the Trailblazer Community for updates.
